rs/node_rewards/canister/src/canister/mod.rs

```diff
@@ -283,19 +285,69 @@ impl NodeRewardsCanister {
         request: GetNodeProviderRewardsCalculationRequest,
     ) -> GetNodeProviderRewardsCalculationResponse {
         let provider_id = ic_base_types::PrincipalId::from(request.provider_id);
-        let request_inner = GetNodeProvidersRewardsRequest {
-            from_nanos: request.from_nanos,
-            to_nanos: request.to_nanos,
-        };
-        let result =
-            canister.with_borrow(|canister| canister.calculate_rewards::<S>(request_inner))?;
-        let node_provider_rewards = result
-            .provider_results
-            .get(&provider_id)
-            .cloned()
-            .ok_or_else(|| format!("No rewards found for node provider {}", provider_id))?;
+        if request.historical {
+            let reward_key = NodeProviderRewardsKey {
```

Comment:
Gven that Governance is minting the rewards, Governance already has to store some of this information.  So we'll want to make sure Governance has what it needs to store these keys when it mints rewards so that it can then reference the rewards that were previously minted.